pkg/dao/consumer.go

```diff
@@ -55,8 +55,12 @@ func (d *sqlConsumerDao) Replace(ctx context.Context, consumer *api.Consumer) (*
 	return consumer, nil
 }
 
-func (d *sqlConsumerDao) Delete(ctx context.Context, id string) error {
+func (d *sqlConsumerDao) Delete(ctx context.Context, id string, unscoped bool) error {
 	g2 := (*d.sessionFactory).New(ctx)
+	if unscoped {
+		// Unscoped is used to permanently delete the record
+		g2 = g2.Unscoped()
+	}
```

Comment:
You need to lock the resource table before deleting. something like:
```
	lockOwnerID, err := s.lockFactory.NewAdvisoryLock(ctx, resource.ID, db.Resources)
	// Ensure that the transaction related to this lock always end.
	defer s.lockFactory.Unlock(ctx, lockOwnerID)
```

Reply:
I found the advisory lock is only used for the resource payload update. Do we need to add this lock to all the write(update,create,delete) operations for the consumer or resource table?